presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -466,6 +466,19 @@ private void destroyIfNecessary()
         return blocked;
     }
 
+    private static boolean isBlocked(Operator operator)
```

Comment:
I'd change this to the following to avoid leaks in the futures:
```java
    private static boolean isOperatorBlocked(Operator operator)
    {
        ListenableFuture<?> operatorBlocked = operator.isBlocked();
        if (!operatorBlocked.isDone()) {
            operatorBlocked.cancel(true);
            return true;
        }
        ListenableFuture<?> waitingForMemory = operator.getOperatorContext().isWaitingForMemory();
        if (!waitingForMemory.isDone()) {
            waitingForMemory.cancel(true);
            return true;
        }
        return false; 
    }
```